src/main/java/de/uhd/ifi/se/decision/management/jira/extraction/RationaleFromCodeCommentExtractor.java

```diff
@@ -0,0 +1,260 @@
+				.collect(Collectors.toList()));
+		TAGS_SEARCH_PATTERN = Pattern.compile(tagSearch, Pattern.CASE_INSENSITIVE);
+		TWO_EMPTY_LINES_PATTERN = Pattern.compile("\\s*\\n\\s*\\n\\s*\\n"); //with optional white spaces
+		SPACE_ATCHAR_LETTER_PATTERN = Pattern.compile("\\s@[a-z]");
+		NEWLINE_CHAR_PATTERN = Pattern.compile("\\n");
+		this.comment = comment;
+		this.elements = new ArrayList<>();
+	}
+
+	public static int getRationaleStartLineInCode(DecisionKnowledgeElement element) {
+		if (!canProcesElement(element)) {
+			return -1;
+		} else {
+			return RationaleCommitElementPositionCodingHelper.getStartLine(element.getKey());
+		}
+	}
+
+	public static int getRationaleEndLineInCode(DecisionKnowledgeElement element) {
+		if (!canProcesElement(element)) {
+			return -1;
+		} else {
+			return RationaleCommitElementPositionCodingHelper.getEndLine(element.getKey());
+		}
+	}
+
+	public static int getRationaleCursorInCodeComment(DecisionKnowledgeElement element) {
+		if (!canProcesElement(element)) {
+			return -1;
+		} else {
+			return RationaleCommitElementPositionCodingHelper.getCursor(element.getKey());
+		}
+	}
+
+	public static boolean canProcesElement(DecisionKnowledgeElement element) {
+		return element.getDocumentationLocation() == DocumentationLocation.COMMIT;
+	}
+
+	public ArrayList<DecisionKnowledgeElement> getElements() {
+		if (comment.commentContent == null || comment.commentContent.trim().equals("")) {
+			return elements;
+		}
+		Matcher tagMatcher = TAGS_SEARCH_PATTERN.matcher(comment.commentContent);
+		int cursorPosition = -1;
+
+		while (tagMatcher.find()
+				&& cursorPosition <= tagMatcher.start()) {
+			cursorPosition = extractElementAndMoveCursor(tagMatcher);
+		}
+		return elements;
+	}
+
+	private int extractElementAndMoveCursor(Matcher tagMatcher) {
+		String rationaleTypeTag = tagMatcher.group();
+		String rationaleType = getRatTypeFromTag(rationaleTypeTag);
+		String rationaleText = comment.commentContent.substring(tagMatcher.end());
+
+		int cursorPosition = tagMatcher.end();
+		int textEnd = getRationaleTextEndPosition(rationaleText);
+		if (textEnd > 0) {
+			rationaleText = rationaleText.substring(0, textEnd);
+		}
+		cursorPosition += rationaleText.length();
+
+		elements.add(addElement(tagMatcher.end(), rationaleText, rationaleType));
+
+		return cursorPosition;
+	}
+
+	private DecisionKnowledgeElement addElement(int start
+			, String rationaleText, String rationaleType) {
+		return new DecisionKnowledgeElementImpl(0
+				, getSummary(rationaleText)
+				, getDescription(rationaleText)
+				, rationaleType.toUpperCase()
+				, "" // unknown, not needed at the moment
+				, calculateAndCodeRationalePositionInSourceFile(start, rationaleText)
+				, DocumentationLocation.COMMIT);
+	}
+
+	private String calculateAndCodeRationalePositionInSourceFile(int start
+			, String rationaleText) {
+		/**
+		 * @issue: what information the dec. know. key encapsulate
```

Comment:
[alternative] I suggest to leave out the colon after the decision knowledge tag! [/alternative] 
[pro] It is not common in JavaDoc to use colons. [/pro]